discord/scheduled_event.py

```diff
@@ -51,10 +53,225 @@
+    year_days: List[:class:`int`]
+        A list of integers representing the year days this event will repeat in.
+
+        This marks the days of the year this event will repeat in, for example, if it
+        is set to `1`, this event will repeat the first day of every year.
+    """
+
+    @overload
+    def __init__(
+        self,
+        start: datetime,
+        *,
+        weekdays: List[Literal[0, 1, 2, 3, 4, 5, 6]],
+        end: Optional[datetime] = ...,
+    ) -> None:
+        ...
+
+    @overload
+    def __init__(
+        self,
+        start: datetime,
+        *,
+        n_weekday: Tuple[Literal[1, 2, 3, 4], int],
+        end: Optional[datetime] = ...,
+    ) -> None:
+        ...
+    
+    @overload
+    def __init__(
+        self,
+        start: datetime,
+        *,
+        month: Literal[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
+        month_days: List[int],
+        end: Optional[datetime] = ...,
+    ) -> None:
+        ...
+
+    @overload
+    def __init__(
+        self,
+        start: datetime,
+        *,
+        year_days: List[int],
+        end: Optional[datetime] = ...,
+    ) -> None:
+        ...
+
+    def __init__(
+        self,
+        start: datetime,
+        *,
+        weekdays: List[Literal[0, 1, 2, 3, 4, 5, 6]] = MISSING,
+        n_weekday: Tuple[Literal[1, 2, 3, 4], int] = MISSING,
+        month: Literal[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12] = MISSING,
+        month_days: List[int] = MISSING,
+        year_days: List[int] = MISSING,
+        end: Optional[datetime] = MISSING,
+    ) -> None:
+        
+        if not start.tzinfo:
+            raise ValueError(
+                '\'start\' must be an aware datetime. Consider using discord.utils.utcnow() or datetime.datetime.now().astimezone() for local time.'
+            )
+        
+        if end not in (MISSING, None):
+            if not end.tzinfo:
+                raise ValueError(
+                    '\'end\' must be an aware datetime. Consider using discord.utils.utcnow() or datetime.datetime.now().astimezone() for local time.'
+                )
+
+        self.start: datetime = start
+        self.end: Optional[datetime] = end if end is not MISSING else None
+
+        self.weekdays: Optional[List[int]] = weekdays if weekdays is not MISSING else None
+        self.n_weekday: Optional[Tuple[int, int]] = n_weekday if n_weekday is not MISSING else None
+        self.month: Optional[int] = month if month is not MISSING else None
+        self.month_days: Optional[List[int]] = month_days if month_days is not MISSING else None
+        self.year_days: Optional[List[int]] = year_days if year_days is not MISSING else None
+        self._interval: int = 1
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, self.__class__):
+            return (
+                self.start == other.start
+            )
+        return NotImplemented
+    
+    def __set_interval(self, value: int) -> None:
```

Comment:
It's uncommon for discord.py to use name mangling on private methods like this. Any particular reason we added it here?